tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenCollectionTest.cs

```diff
@@ -67,5 +80,29 @@ private void ParseCollectionOptions(BsonDocument document)
                 _collection = _collection.WithWriteConcern(writeConcern);
             }
         }
+
+        private void ParseDatabaseOptions(BsonDocument document, out IMongoDatabase database)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(document, "readConcern", "readPreference", "writeConcern");
+
+            database = _collection.Database;
+            if (document.Contains("readConcern"))
+            {
+                var readConcern = ReadConcern.FromBsonDocument(document["readConcern"].AsBsonDocument);
+                database = database.WithReadConcern(readConcern);
+            }
+
+            if (document.Contains("readPreference"))
+            {
+                var readPreference = ReadPreference.FromBsonDocument(document["readPreference"].AsBsonDocument);
+                database = database.WithReadPreference(readPreference);
+            }
+
+            if (document.Contains("writeConcern"))
+            {
+                var writeConcern = WriteConcern.FromBsonDocument(document["writeConcern"].AsBsonDocument);
+                database = database.WithWriteConcern(writeConcern);
+            }
```

Comment:
Update `_collection` here:

            // update _collection to be associated with the reconfigured database
            _collection = database.GetCollection<BsonDocument>(
                _collection.CollectionNamespace.CollectionName,
                _collection.Settings);